docs/specification/draft/basic/streams.mdx

```diff
@@ -0,0 +1,242 @@
+---
+title: Streams
+---
+
+<div id="enable-section-numbers" />
+
+<Info>**Protocol Revision**: draft</Info>
+
+MCP defines a transport-agnostic mechanism for resumable streams, allowing clients and servers to maintain state across disconnections.
+
+## Introduction
+
+Streams provide a way to manage long-running operations and ensure message delivery even when connections are interrupted. Key benefits include:
+
+- Transport-agnostic resumability (works with HTTP, WebSocket, stdio, etc.)
+- Ability for clients and servers to disconnect and reconnect without losing progress
+- Resource management through configurable timeouts
+- Status checking without requiring full reconnection
+
+## Lifecycle
+
+A stream follows a defined lifecycle:
+
+1. **Creation**: The server creates a stream and sends a `notifications/stream/create` notification with a unique stream ID
+2. **Live**: Messages are exchanged on the stream.
+3. **Completion**: Server sends a `notifications/stream/end` notification when the stream is complete
+
+A client may use the `stream/resume` request to receive messages from a live stream, and `stream/poll` to poll a stream's state. A client may also use `stream/poll/all` to poll all available streams.
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant Server
+
+    Note over Client,Server: Stream Creation
+    Client->>+Server: Request (e.g., tools/call)
+    Server-->>Client: notifications/stream/create
+
+    Note over Client,Server: Stream Operation
+    loop Message Exchange
+        Server-->>Client: Messages on stream
+    end
+
+    Note over Client,Server: Stream Termination
+    Server-->>-Client: notifications/stream/end
+```
+
+### Creation
+
+When a client sends a request that may produce a long-running response, the server can create a stream by sending a `notifications/stream/create` notification with a unique stream ID and resume interval parameters.
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "notifications/stream/create",
+  "params": {
+    "stream": {
+      "streamId": "550e8400-e29b-41d4-a716-446655440000",
+      "resumeInterval": {
+        "min": 5,
+        "max": 3600
+      }
+    }
+  }
+}
+```
+
+The `resumeInterval` specifies:
+
+- `min`: Minimum seconds a client should wait before resuming the stream (prevents excessive reconnections)
+- `max`: Maximum seconds a client may wait before the server considers the stream abandoned
```

Comment:
Are the Min and Max arguments mandatory here? How can I represent a stream that should never be considered abandoned?

Reply:
The arguments are not mandatory: https://github.com/modelcontextprotocol/modelcontextprotocol/pull/899/files#diff-273099cbfa5ca22c6af97f73eca738047952f7a8e7dffc4c1572fec3bdbbc211R1408-R1422.  So you can omit `max` to indicate that a stream will never be considered abandoned.  We can make a note of that in the docs.